### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: index.js
Comment (andrewwa-kainos) [inline index.js:31]:
```suggestion
    'class-methods-use-this': 'off', // Stops us from having to declare class methods which don't use this as static.
```
minor: typo fix

**Comment by Cavalier07** [comment]
Could we make sure we get an approval from the DVSA Architects before merging :) Last change we made for FTTS had to go through Nick Lee, Tony Smith and Shaun Hare before we merged

**Review by sdh100shaun — APPROVED** [review]

**Comment by sdh100shaun** [comment]
thanks @Cavalier07  yes I am happy to update these ,  from FTTS side we will tag and publish if you thumbs up too

File: index.js
Comment (andytaylorkainos) [reply index.js]: fixed

**Comment by Cavalier07** [comment]
Hi,

Just ran the new rules against one of our smaller azure functions and got 35000 issues reported. The big offender is on the rule:

"@typescript-eslint/camelcase" 

Due to the naming conventions in Dynamics CRM we have to use underscores in variables names such as ftts_name, so we would have to disable that rule across the project.

The other ones which cropped up are 

@typescript-eslint/no-redeclare
@typescript-eslint/no-shadow

We get the following error for those: `Definition for rule 'xyz' was not found`, I've not looked any more deeply into that yet.

There are a few other triggers we get around naming for enums, and unused variables.

On publishing, if those are the rules that DVSA want to enforce I don't see a major issue from this quick look and am happy to give it a green tick. There will have to be tickets on FTTS to a more in-depth assessment and then upgrade to the new version across the codebase.

Tom

RE: @janmrozowicz

**Review by Cavalier07 — APPROVED** [review]

**Comment by andytaylorkainos** [comment]
I've discovered that, when extending this config into another project, `@typescript-eslint/eslint-plugin` must be included within that project's dependencies and it must match the same version of that in this repository's `package.json` otherwise they conflict.

If this PR causes more trouble than it's worth with other projects, it might be best to keep the versions as they are? @sdh100shaun

**Comment by Cavalier07** [comment]
@andytaylorkainos - Could you add that to the Readme?

**Comment by andytaylorkainos** [comment]
@Cavalier07, @sdh100shaun  - I have finally managed to resolve all of the issues that I previously mentioned without having to update the dependencies within this repository or include any of those dependencies in the project that imports this plugin.

It may be worth updating these dependencies at some point but it will, as @Cavalier07  has said, cause issues as the standards have changed between versions and this will directly affect all projects that extend this configuration.

**Comment by Cavalier07** [comment]
@andytaylorkainos  I'm a fan of updating as soon as possible. It may cause us a couple of days of pain, but the longer we leave it the worse it will be.

**Comment by sdh100shaun** [comment]
agreed with @Cavalier07 let's get this updated

**Comment by andytaylorkainos** [comment]
Awesome, if someone could merge and tag it for release that'd be great :)

**Comment by Cavalier07** [comment]
@andytaylorkainos - Merged do you need it published up to NPM or can you do that yourself

**Comment by andytaylorkainos** [comment]
@Cavalier07, i dont think I have access - if you could do it for me, it'd be a huge help :)

**Comment by Cavalier07** [comment]
@andytaylorkainos - Have published it as v2.0.0 due to the issues which may be caused by the upgrade

Could you approve this PR so I can merge the new version into master - https://github.com/dvsa/eslint-config-ts/pull/5

**Pull request merged**
